src/BenchmarkDotNet/Portability/RuntimeInformation.cs

```diff
@@ -26,6 +28,9 @@ internal static class RuntimeInformation
 
         public static bool IsMono { get; } = Type.GetType("Mono.Runtime") != null; // it allocates a lot of memory, we need to check it once in order to keep Engine non-allocating!
 
+#if NET6_0_OR_GREATER
+        [System.Runtime.Versioning.SupportedOSPlatformGuard("windows")]
+#endif
         public static bool IsFullFramework => FrameworkDescription.StartsWith(".NET Framework", StringComparison.OrdinalIgnoreCase);
```

Comment:
This could be always false on .NET 6+. I'm not sure we can optimize other `Is***` methods, so let's be careful.

Reply:
Again, this was just to fix-up new compatibility warnings. How `IsFullFramework` is implemented being changed would be a separate concern to adding a new TFM?

Reply:
Ah I see what you mean now. By definition, if it's `NET6_0_OR_GREATER` then the implementation is just `return false;` because it isn't .NET Framework.